### PR TITLE
DDT-672: PUT response payload

### PR DESCRIPTION
**Pull request by Abdiiir**
- updated body response & unit tests

### CODE REVIEW

File: src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java [hunk 1]
```diff
@@ -703,6 +702,7 @@ private BookingTO bookingToDTOWithNullLocations(Booking booking) {
             b -> {
               // update booking with new booking request
               Booking booking = bookingMapper.dtoToBooking(bookingRequest);
+              booking.setUpdatedDateTime(updatedDateTime);
```
Comment (douradofrazer) [inline src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:705]: I did the same fix in a PR before this that was merged, it's weird that it does not conflict.

Comment (Abdiiir) [reply src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:705]: Yeah, I looked into this and can see your point. Not sure why this happened, but the code is not duplicated

**Review by douradofrazer — DISMISSED** [review]

File: src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java (hunk 1 above)
Comment (douradofrazer) [reply src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:705]: @Abdiiir make sure you verify that the code isn't overwritten on merge.

Comment (Abdiiir) [reply src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java]: This should be resolved now and the duplicate line caused from re-basing is removed. 

@douradofrazer waiting for a second review.

File: src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java [hunk 2]
```diff
@@ -688,10 +688,9 @@ private BookingTO bookingToDTOWithNullLocations(Booking booking) {
 
   @Override
   @Transactional
-  public Mono<BookingTO> updateBookingByReferenceCarrierBookingRequestReference(
-      String carrierBookingRequestReference, BookingTO bookingRequest) {
-
-    if (!carrierBookingRequestReference.equals(
+  public Mono<BookingResponseTO> updateBookingByReferenceCarrierBookingRequestReference(String carrierBookingRequestReference, BookingTO bookingRequest) {
```
Comment (mp-assco) [inline src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:691]: I think we need some checks similar to what we have at the beginning of `createBooking` concerning `getIsImportLicenseRequired`, `getImportLicenseReference`, `getIsExportDeclarationRequired`, `getExportDeclarationReference`.

Comment (Abdiiir) [reply src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:691]: Yes, will include that.

File: src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java [hunk 3]
```diff
@@ -688,26 +688,29 @@ private BookingTO bookingToDTOWithNullLocations(Booking booking) {
...
+      if (!carrierBookingRequestReference.equals(
         bookingRequest.getCarrierBookingRequestReference())) {
       return Mono.error(
           new UpdateException("carrierBookingRequestReference in path does not match body."));
     }
+      if (bookingRequest.getIsImportLicenseRequired()
```
Comment (douradofrazer) [inline src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:697]: This looks like duplicate code, extract it out ?

File: src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java [hunk 4]
```diff
@@ -1515,15 +1519,32 @@ private BookingTO bookingToDTOWithNullLocations(Booking booking) {
...
         if (allowedDocumentStatuses.contains(booking.getDocumentStatus())) {
           return Mono.just(booking);
         } else
           return Mono.error(
               new UpdateException(
-                  "Cannot Cancel Booking that is not in status RECE, PENU or CONF"));
+                  "Cannot Cancel Booking that is not in status RECE, PENU, CONF or PENC"));
       };
+
+    private final Function<Booking, Mono<Booking>> checkUpdateBookingStatuses =
```
Comment (douradofrazer) [inline src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:1534]: I think we can consolidate this with the above method and keep the check on only `allowedDocumentStatuses` I don't think there is a specific need to check for `CANC`, @pimg what do you think?

**Review by douradofrazer — CHANGES_REQUESTED** [review]

File: src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java (hunk 4 above)
Comment (Abdiiir) [reply src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:1534]: Do we not then somehow allow for cancelled bookings to be updated?

Comment (HenrikHL) [reply src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:1534]: I am quite sure Christian has said that once a Booking is Cancelled - then it cannot be modified... @pimg please also confirm...

Comment (pimg) [reply src/main/java/org/dcsa/bkg/service/impl/BookingServiceImpl.java:1534]: correct cancelled is an end state

**Review by douradofrazer — APPROVED** [review]
I do see some duplications in checks for update and post while reviewing the PR's, Will be a good idea to refactor once merged.

**Pull request merged**